dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/Assertions.java

```diff
@@ -210,9 +226,11 @@ public static void assertHasWarning( ValidationReport report, Warning warning )
     {
         assertTrue( report.hasWarnings(), "warning not found since report has no warnings" );
         assertTrue( report.hasWarning( w -> Objects.equals( warning.getWarningCode(), w.getWarningCode() ) &&
-            Objects.equals( warning.getUid(), w.getUid() ) ),
-            String.format( "warning with code %s for object %s not found in report with warning(s) %s",
-                warning.getWarningCode(), warning.getUid(), report.getWarnings() ) );
+            Objects.equals( warning.getUid(), w.getUid() ) &&
+            w.getWarningMessage().contains( warning.getWarningMessage() ) ),
```

Comment:
This also adds asymmetry to our API. We are now asserting on the warning message in `assertHasWarning` but not the error message in `assertHasError`.